xdoc/file_read_text.rst

```diff
@@ -0,0 +1,187 @@
+Reading Text from a File
+========================
+
+* there are 9 different variations showing reading a text file
+* this is not an exhaustive list of how to read a text file
+* each reading of the input file is timed
+* each file reading is approximately ordered from slower to faster
+* if the examples were to manipulate strings within the loop, then the overall times would be much slower
+* Garbage collection and OS load will add variability to the timing (this is NOT a benchmark)
+* Provide some suitably sized (text) file called junkdata.txt when trying this example
+* compile with -d:release (the file reading is slower otherwise)
+  
+
+.. code-block:: nim
+  
+  import strutils, times, os, parsecsv, streams, memfiles
+  
+  let
+    fname = "junkdata.txt"  
+
+  const
+    FGETSZ = 2500    
+    
+  var
+    t0, t1 = cpuTime()
+    F: File
+    line = ""
+    cntL = 0
+    cntC = 0
+  
+  var fgetStr: cstring = newString(FGETSZ+1)
+  
+  proc fgets(c: cstring, n: cint, f: File): cstring  {.importc: "fgets", header: "<stdio.h>".}
```

Comment:
At least add a warning here that ``fgets`` is really a horrible thing and one should avoid it at all costs.